dev/devicelab/bin/tasks/new_gallery_web_compile_test.dart

```diff
@@ -0,0 +1,90 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:flutter_devicelab/framework/framework.dart';
+import 'package:flutter_devicelab/framework/utils.dart';
+
+Future<void> main() async {
+  await task(const NewGalleryWebCompileTest().run);
+}
+
+/// Measures the time to compile the New Flutter Gallery to JavaScript
+/// and the size of the compiled code.
+class NewGalleryWebCompileTest {
```

Comment:
There should already be some compile size tasks for the web, it would be worth sharing some of that logic if possible

Reply:
I'm looking at it now.

Reply:
I did some refactoring.